subroutines.sh

```diff
@@ -132,8 +164,18 @@ listen_to_ble() {
       MQTT_TOPIC="tesla_ble/$VIN/binary_sensor/presence"
 
       # Check the presence using both MAC Addr and BLE Local Name
-      log_debug "$(echo "$BLTCTL_OUT" | grep -E "($BLE_MAC|$BLE_LN)")"
-      check_presence "BLE MAC & LN" "($BLE_MAC|$BLE_LN)"
+      log_debug "BLTCTL_OUT:$(echo "$BLTCTL_OUT" | grep -E "($BLE_MAC|$BLE_LN)")"
+      macAddr=$(check_presence "BLE LN & MAC" $BLE_LN $BLE_MAC)
+
+      # Add the MAC address to BLE_MAC_LIST if it's not already present at the position
+      if [ "$macAddr" != "$BLE_MAC" ]; then
+        eval "BLE_MAC_LIST=\$(echo \$BLE_MAC_LIST | awk '{\$${position}=\"$macAddr\"; print}')"
+        log_debug "listen_to_ble; BLE_MAC_LIST:$BLE_MAC_LIST"
+        [ ! -f $KEYS_DIR/${vin}_macaddr ] && echo $macAddr >$KEYS_DIR/${vin}_macaddr
```

Comment:
good
Then do you check the presence of this file somewhere? Did not find it.

Reply:
Yes in run.sh `for loop` - it adds either the real MAC address or "fake" ``FF:FF...FF`.